src/sdp/RemoteSdp.cpp

```diff
@@ -175,21 +229,79 @@ namespace mediasoupclient
 		return sdptransform::write(this->sdpObject);
 	}
 
-	void Sdp::RemoteSdp::AddMediaSection(MediaSection* mediaSection)
+	void Sdp::RemoteSdp::AddMediaSection(MediaSection* newMediaSection)
+	{
+		MSC_TRACE();
+
+		if (this->firstMid.empty())
+			this->firstMid = newMediaSection->GetMid();
+
+		// Add it in the vector.
+		this->mediaSections.push_back(newMediaSection);
+
+		// Add to the map.
+		this->midToIndex[newMediaSection->GetMid()] = this->mediaSections.size() - 1;
+
+		// Add to the SDP object.
+		this->sdpObject["media"].push_back(newMediaSection->GetObject());
+
+		this->RegenerateBundleMids();
+	}
+
+	void Sdp::RemoteSdp::ReplaceMediaSection(MediaSection* newMediaSection, const std::string& reuseMid)
 	{
 		MSC_TRACE();
 
 		// Store it in the map.
-		this->mediaSections[mediaSection->GetMid()] = mediaSection;
+		if (!reuseMid.empty())
+		{
+			const auto idx = this->midToIndex[reuseMid];
+			const auto oldMediaSection = this->mediaSections[idx];
+
+			// Replace the index in the vector with the new media section.
+			this->mediaSections[idx] = newMediaSection;
+
+			// Update the map.
+			this->midToIndex.erase(oldMediaSection->GetMid());
+			this->midToIndex[newMediaSection->GetMid()] = idx;
+
+			// Delete old MediaSection.
+			delete oldMediaSection;
 
-		// Update SDP object.
-		this->sdpObject["media"].push_back(mediaSection->GetObject());
+			// Update the SDP object.
+			this->sdpObject["media"][idx] = newMediaSection->GetObject();
 
-		std::string mids = this->sdpObject["groups"][0]["mids"].get<std::string>();
-		if (mids.empty())
-			mids = mediaSection->GetMid();
+			// Regenerate BUNDLE mids.
+			this->RegenerateBundleMids();
+		}
 		else
-			mids.append(" ").append(mediaSection->GetMid());
+		{
+			const auto idx = this->midToIndex[newMediaSection->GetMid()];
+
+			// Replace the index in the vector with the new media section.
+			this->mediaSections[idx] = newMediaSection;
```

Comment:
**TODO:** This is leaking the old `MediaSection*` that was in `this->mediaSections[idx]` before replacing its value with the new one.